utils/generate-stack.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+set -e
+
+if [ -z "${STACK_NAME}" ]; then
+  echo "STACK_NAME variable not defined."
+  exit 1
+fi
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+
+# create stack directory
+NEW_DIR_PATH="$ROOT_DIR/stacks/$STACK_NAME"
+mkdir "$NEW_DIR_PATH"
+
+# copy files
+cp -r "$ROOT_DIR"/utils/stack-templates/* "$NEW_DIR_PATH"
+
+for file in "$NEW_DIR_PATH"/*
```

Comment:
why are we putting the yaml into the stack? doesn't this mean that if we want to use this yaml in any other stack will have to also have it in that stack, thus causing duplication all over? my understanding was that we'd have one place were the yaml lived and reference it from within a stack. that way if we update the yaml we do it in one place, versus doing it in each stack that we create.

Reply:
There's src YAML (typically included as part of a Helm template, or raw k8s manifests), and stack (rendered) YAML. Rendered meaning it has either been customized using Kustomize, Helm values, or both.

Reply:
👍